sdk/keyvault/azcertificates/client.go

```diff
@@ -0,0 +1,1568 @@
+
+	// The management policy for the certificate.
+	CertificatePolicy *CertificatePolicy `json:"policy,omitempty"`
+
+	// Application specific metadata in the form of key-value pairs.
+	Tags map[string]string `json:"tags,omitempty"`
+}
+
+func (u *UpdateCertificatePropertiesOptions) toGenerated() *generated.KeyVaultClientUpdateCertificateOptions {
+	return &generated.KeyVaultClientUpdateCertificateOptions{}
+}
+
+// UpdateCertificatePropertiesResponse contains the result from method Client.UpdateCertificateProperties.
+type UpdateCertificatePropertiesResponse struct {
+	CertificateBundle
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+}
+
+// UpdateCertificate applies the specified update on the given certificate; the only elements updated are the certificate's
+// attributes. This operation requires the certificates/update permission.
+func (c *Client) UpdateCertificateProperties(ctx context.Context, certName string, options *UpdateCertificatePropertiesOptions) (UpdateCertificatePropertiesResponse, error) {
+	if options == nil {
+		options = &UpdateCertificatePropertiesOptions{}
+	}
+	resp, err := c.genClient.UpdateCertificate(
+		ctx,
+		c.vaultURL,
+		certName,
+		options.Version,
+		generated.CertificateUpdateParameters{
+			CertificateAttributes: options.CertificateAttributes.toGenerated(),
+			CertificatePolicy:     options.CertificatePolicy.toGeneratedCertificateCreateParameters(),
+			Tags:                  convertToGeneratedMap(options.Tags),
+		},
+		options.toGenerated(),
+	)
+	if err != nil {
+		return UpdateCertificatePropertiesResponse{}, err
+	}
+	return UpdateCertificatePropertiesResponse{
+		RawResponse:       resp.RawResponse,
+		CertificateBundle: certificateBundleFromGenerated(&resp.CertificateBundle),
+	}, nil
+}
+
+// MergeCertificateOptions contains the optional parameters for the Client.MergeCertificate function.
+type MergeCertificateOptions struct {
+	// The attributes of the certificate (optional).
+	CertificateAttributes *CertificateAttributes `json:"attributes,omitempty"`
+
+	// Application specific metadata in the form of key-value pairs.
+	Tags map[string]string `json:"tags,omitempty"`
+}
+
+func (m *MergeCertificateOptions) toGenerated() *generated.KeyVaultClientMergeCertificateOptions {
+	return &generated.KeyVaultClientMergeCertificateOptions{}
+}
+
+// MergeCertificateResponse contains the response from method Client.MergeCertificate.
+type MergeCertificateResponse struct {
+	CertificateBundle
+
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+}
+
+// The MergeCertificate operation performs the merging of a certificate or certificate chain with a key pair currently available in the service. This operation requires the certificates/create permission.
+func (c *Client) MergeCertificate(ctx context.Context, certName string, certificates [][]byte, options *MergeCertificateOptions) (MergeCertificateResponse, error) {
```

Comment:
We pack every parameter into a `MergeCertificateOptions`: https://github.com/Azure/azure-sdk-for-net/blob/a7743d7564c9d60dff0ec17822af0f18323f5f15/sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs#L1606

No other parameters are exposed (sans pipeline options, etc.).

Reply:
The Certificates are required parameters correct, can you pass a nil parameter and have a successful response from the service? If you can then you're correct we can pack it into the options. `certName` is also a required parameter so we cannot  pack it into the options. The thought being, a user should always be able to pass in `nil` for an Options struct for the default options.

Reply:
"options" doesn't mean optional, per architects on past reviews. For a brief point in time, they wanted to use *Parameters for any options bag that had required parameters (hence our `CryptographyClient` method signautres), but changed to just use *Options for everything - required or not. So in this case, `MergeCertificateOptions` is required and has required fields, but also optional fields. That's what we do in other languages as well.

Reply:
As Go doesn't have default/optional parameters, we have a `<ClientName><Method>Options` type that contains all non-mandatory parameters.  All methods take this regardless of if they contain optional parameters.  All mandatory parameters are pass individually.  This is the pattern all of our SDKs follow, please see the [guidelines](https://azure.github.io/azure-sdk/golang_introduction.html) for more info.